src/binary/non_blocking/binary_buffer.rs

```diff
@@ -93,7 +93,7 @@ impl<A: AsRef<[u8]>> BinaryBuffer<A> {
     /// If the buffer is not empty, returns `Some(_)` containing the next byte in the buffer.
     /// Otherwise, returns `None`.
     pub fn peek_next_byte(&self) -> Option<u8> {
-        self.bytes().get(0).copied()
+        self.data.as_ref().get(self.start).copied()
```

Comment:
🗺️ The original code created a byte slice (`&[u8]`) even though we only needed the first byte in the data. Now it just gets the first byte without creating an intermediate slice.